app/scripts/modules/core/src/pipeline/config/stages/stage.html

```diff
@@ -64,6 +64,26 @@ <h4 ng-bind="stage.name || '[new stage]'"></h4>
           </ui-select-choices>
         </ui-select>
       </stage-config-field>
+      <stage-config-field label="Authorized Groups" help-key="pipeline.config.trigger.runAsUser" ng-if="stage.type === 'manualJudgment'" style="margin-bottom: 10px">
+        <ui-select
+          ng-model="stage.selectedStageRoles"
+          multiple
+          class="form-control input-sm"
+          on-select="updateAvailableStageRoles()"
+          on-remove="updateAvailableStageRoles()"
+        >
+          <ui-select-match>{{$item.name}}</ui-select-match>
+          <ui-select-choices
+            repeat="option.roleId as option in options.stageRoles | anyFieldFilter: {name: $select.search}"
+            ui-disable-choice="!option.available"
+          >
+            <span
+              ng-if="!stage.selectedStageRoles.includes(option.roleId)"
+              ng-bind-html="option.name | highlight: $select.search"
+            ></span>
+          </ui-select-choices>
+        </ui-select>
+      </stage-config-field>
```

Comment:
> What the reasoning behind adding the new Authorized Groups UI to the base `stage.html` and only displaying it if the stage being rendered is a Manual Judgment stage, rather than just adding the new Authorized Groups UI to the [Manual Judgment stage](https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/pipeline/config/stages/manualJudgment/manualJudgmentStage.js) itself? Do you have plans to extend this feature to other stages? If not, can we move this template and corresponding controller logic into the Manual Judgment module?

Yes we have plans to extend this feature to other stages.

Reply:
> Yes we have plans to extend this feature to other stages.

Thanks for letting me know; I'm curious what other stage types we think this feature would make sense for. Are there stage types other than Manual Judgment that require mid-pipeline user input for which the roles of the approving user would be relevant?